### PR TITLE
[Feat] Step charts and Plop fixes

### PR DESCRIPTION
**Pull request by speckospock**
Adds Step chart component.

## Description

Creates a new Step component, using the Plop workflow. Using Plop workflow highlighted a bug preventing the documentation from rendering correctly, so that bug is also fixed (it required ignoring src/charts/template in the styleguide.config.js file)

## Motivation and Context

Moves closer to 1:1 parity between Britecharts and Britecharts-React

## How Has This Been Tested?

Per the Contributing guide, tests were implemented through modifying the Plop boilerplate, and double-checked by confirming visual accuracy in the documentation. Full test suite passes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/docs/styles/custom.css [hunk 1]
```diff
@@ -25,6 +25,15 @@ body,
     display: flex;
 }
 
+.step-container {
+    display: flex;
+    padding: 2%;
```
Comment (Golodhros) [inline src/docs/styles/custom.css]: Interesting, what are these styles fixing?

Comment (speckospock) [reply src/docs/styles/custom.css]: The step chart, when rendered in the documentation, spilled slightly out of the container. So, this was to bring it back into bounds.

Comment (Golodhros) [reply src/docs/styles/custom.css]: I see, maybe the default size is larger than the container?

Could you check if using the responsive container HOC --like in https://github.com/eventbrite/britecharts-react/blob/master/src/charts/bar/Readme.md-- this gets solved?

This way we could avoid including this custom styles.
I would avoid this styles and use the responsive HOC for this,

Comment (speckospock) [reply src/docs/styles/custom.css]: I implemented the fix as you suggested, and it works! Thanks for pointing me in the right direction.

**Pull request merged**